apps/prairielearn/src/lib/question-render.ts

```diff
@@ -586,27 +640,37 @@ export async function renderPanelsForSubmission({
           instance_question_id: variant.instance_question_id,
         });
 
-  /** @type {SubmissionPanels} */
-  const panels = {
+  const group_role_permissions = await run(async () => {
+    if (!instance_question || !assessment_instance?.group_id || !group_config?.has_roles) {
+      return null;
+    }
+
+    return await getQuestionGroupPermissions(
+      instance_question?.id,
+      assessment_instance?.group_id,
+      user_id,
+    );
+  });
```

Comment:
While I'm pretty sure I can prove to myself that this is true, I'm a little hesitant to make that query conditional until we're no longer passing around a huge `locals` object everywhere. `group_role_permissions` ends up being used in `buildLocals`, and we currently don't have any strong guarantees that the locals derived from `group_role_permissions` aren't used as part of rendering submissions or their panels. Does that make sense?

Reply:
I understand. For what it's worth, the locals updated in buildLocals from this conditional are only used in questionFooter. That said, I'm ok with keeping the query always there for now.

Reply:
Also understood. That's true _right now_, but it would be really easy to accidentally change that in the future. I'll leave this as-is until we can hopefully get this even more precisely typed and kill `locals` completely.